File: src/main/java/org/aldo/beautycenter/data/dto/UpdateServiceDto.java
package org.aldo.beautycenter.data.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import org.aldo.beautycenter.security.customAnnotation.annotation.ValidMultipartExtension;
import org.aldo.beautycenter.security.customAnnotation.annotation.ValidServiceId;
import org.springframework.web.multipart.MultipartFile;

@Data
public class UpdateServiceDto {
    @NotNull(message = "Il campo id è obbligatorio")
    @ValidServiceId
    private String id;

    @ValidMultipartExtension
    private MultipartFile image;

    @NotBlank(message = "Il campo name è obbligatorio")
    @Size(min = 1, max = 50, message = "Il campo name deve essere compreso tra 1 e 50 caratteri")
    private String name;

    @NotNull(message = "Il campo prezzo è obbligatorio")
    @Min(value = 1, message = "Il campo durata dev'essere almeno 1")
    private Double price;

    @NotNull(message = "Il campo durata è obbligatorio")
    @Min(value = 1, message = "Il campo durata dev'essere almeno 1")
    private Long duration;
}

File: lib/model/UpdateServiceDto.dart
import 'dart:io';

// todo rimuovere

class UpdateServiceDto {
  final String id;
  final File? image;
  final String name;
  final double price;
  final int duration;

  UpdateServiceDto({
    required this.id,
    this.image,
    required this.name,
    required this.price,
    required this.duration,
  });

  UpdateServiceDto copyWith({
    String? id,
    File? image,
    String? name,
    double? price,
    int? duration,
  }) {
    return UpdateServiceDto(
      id: id ?? this.id,
      image: image ?? this.image,
      name: name ?? this.name,
      price: price ?? this.price,
      duration: duration ?? this.duration,
    );
  }
}